README.md

```diff
@@ -257,6 +216,44 @@ try {
 
 See [API documentation](docs/api.md) for more example and full API reference information.
 
+## Generator version vs Template version
+
+The generator is a tool that you can use to generate whatever you want, taking AsyncAPI specification file as an input. A template is a tool that uses Generator features and helpers to specify what should be generated.
+
+In other words, a template depends on the generator and its features. For example it might work with the latest version of the generator but not the previous ones. 
```

Comment:
```suggestion
In other words, a template depends on the Generator and its features. For example, it might work with the latest version of the Generator but not the previous ones. 
```